buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -596,6 +596,7 @@ class BeamModulePlugin implements Plugin<Project> {
         'options',
         'cast',
         'deprecation',
+        'fallthrough',
```

Comment:
I don't see a new case where this is needed in this PR. We have pinned error-prone to prevent new warnings from hitting us. I'm inclined to keep the warning and suppress on a case-by-case. Can you clarify what is the reason here?

Reply:
This is the configuration for javac lint, not error-prone. javac lint analyzes generated code and calcite generates code that makes it fail.